src/fairing.rs

```diff
@@ -122,19 +122,60 @@ impl Fairing for Compression {
 ///
 /// rocket::build()
 ///     // ...
-///     .attach(CachedCompression::fairing(vec![".otf", "main.dart.js"]))
+///     .attach(CachedCompression {
+///       cached_paths: vec!["", "/", "/about", "/people", "/posts", "/events", "/groups"],
+///       cached_path_prefixes: vec!["/user/", "/g/", "/p/"],
+///       cached_path_suffixes: vec![".otf", "main.dart.js"]
+///     })
 ///     // ...
 ///     # ;
 ///
 /// ```
 pub struct CachedCompression {
-    pub cached_path_endings: Vec<&'static str>,
+    pub cached_paths: Vec<&'static str>,
+    pub cached_path_prefixes: Vec<&'static str>,
+    pub cached_path_suffixes: Vec<&'static str>,
+    pub excluded_path_prefixes: Vec<&'static str>,
 }
 
 impl CachedCompression {
-    pub fn fairing(cached_path_endings: Vec<&'static str>) -> CachedCompression {
+    /// Caches only the specific paths provided.
+    pub fn specific_fairing(cached_paths: Vec<&'static str>) -> CachedCompression {
```

Comment:
I'd recommend renaming this to `exact_path_fairing`.  Although this new comment is good, I think that name would make it more clear what this does.